src/dstack/_internal/server/services/encryption/keys/aes.py

```diff
@@ -0,0 +1,62 @@
+import os
+from base64 import b64decode, b64encode
+from typing import Literal
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+from pydantic import Field, validator
+from typing_extensions import Annotated
+
+from dstack._internal.core.models.common import CoreModel
+from dstack._internal.server.services.encryption.keys.base import EncryptionKey
+
+
+class AESEncryptionKeyConfig(CoreModel):
+    type: Annotated[Literal["aes"], Field(description="The type of the key")] = "aes"
+    secret: Annotated[str, Field(description="Base64-encoded AES-256 key")]
+
+    @validator("secret")
+    def validate_secret(cls, v):
+        try:
+            key = b64decode(v, validate=True)
+        except Exception as e:
+            raise ValueError("Failed to decode secret from base64") from e
+        if len(key) != 32:
+            raise ValueError(f"AES key must be 32 bytes. Got {len(key)} bytes")
+        return v
+
+
+class AESEncryptionKey(EncryptionKey):
+    TYPE = "aes"
+
+    def __init__(self, config: AESEncryptionKeyConfig) -> None:
+        self.key = b64decode(config.secret)
+
+    def encrypt(self, plaintext: str) -> str:
+        # Generate a random 16-byte (128-bit) IV
+        iv = os.urandom(16)
+
+        # Create AES-GCM Cipher object and encrypt
+        cipher = Cipher(algorithms.AES(self.key), modes.GCM(iv), backend=default_backend())
```

Comment:
Did you consider using [`aead.AESGCM`](https://cryptography.io/en/latest/hazmat/primitives/symmetric-encryption/#cryptography.hazmat.primitives.ciphers.modes.GCM) (with `associated_data=None`) instead of `modes.GCM`? I'm not sure if it would change anything apart from the interface, but the [docs](https://cryptography.io/en/latest/hazmat/primitives/symmetric-encryption/#cryptography.hazmat.primitives.ciphers.modes.GCM) suggest doing so:
> If you are encrypting data that can fit into memory you should strongly consider using AESGCM instead of this.

Reply:
I began working on the AES implementation so that it can be extended to support different modes. Probably makes sense to switch to [aead.AESGCM](https://cryptography.io/en/latest/hazmat/primitives/symmetric-encryption/#cryptography.hazmat.primitives.ciphers.modes.GCM) to simplify the code.

Reply:
Moved to [aead.AESGCM](https://cryptography.io/en/latest/hazmat/primitives/symmetric-encryption/#cryptography.hazmat.primitives.ciphers.modes.GCM)